secantTests/LoadingScreenTests.swift

```diff
@@ -0,0 +1,239 @@
+//
+//  AppRouterNavigationTests.swift
+//  secantTests
+//
+//  Created by Francisco Gindre on 9/2/21.
+//
+
+import XCTest
+@testable import secant_testnet
+import Foundation
+import Combine
+class LoadingScreenTests: XCTestCase {
+    var cancellables: [AnyCancellable] = []
+    
+    // MARK: LoadingScreenViewModel Tests
+    
+    func testLoadingPublishedHomeScreenWhenCredentialsArePresent() throws {
+        let mockServices = MockServices()
+        let stub = KeysPresentStub(returnBlock: {
+            true
+        })
+        mockServices.keyStorage = stub
+        let loadingViewModel = LoadingScreenViewModel(services: mockServices)
+        // swiftlint:disable:next line_length
+        let testExpectation = XCTestExpectation(description: "LoadingViewModel Publishes .credentialsFound when credentials are present and there's no failure")
+        let expected = LoadingScreenViewModel.LoadingResult.credentialsFound
+        
+        loadingViewModel.$loadingResult
+            .dropFirst()
+            .sink { loadingResult in
+                testExpectation.fulfill()
+                
+                XCTAssertTrue(stub.called)
+                switch loadingResult {
+                case .success(let result):
+                    XCTAssertEqual(result, expected)
+                case .failure(let error):
+                    XCTFail("found error \(error.localizedDescription)")
+                case .none:
+                    XCTFail("found None when expected a value")
+                }
+            }
+            .store(in: &cancellables)
+        loadingViewModel.loadAsync()
+        wait(for: [testExpectation], timeout: 0.1)
+    }
+
+    func testLoadingPublishesNavigatesToOnboardingWhenNoCredentialsFound() throws {
```

Comment:
This test reads to me like we're waiting for the view model to publish newWallet, which may end up having something to do with navigating but should the test description just call out what we are asserting?  This applies to most of these tests here I believe.